Legion/CourtOfStars/AdvisorMelandrus.lua

```diff
@@ -93,6 +95,7 @@ do
 end
 
 function mod:SlicingMaelstrom(args)
-	self:MessageOld(args.spellId, "yellow", "warning", CL.incoming:format(args.spellName))
-	self:CDBar(args.spellId, 24)
+	self:MessageOld(args.spellId, "yellow", "warning",  CL.count:format(args.spellName, slicingMaelstromCount))
+	slicingMaelstromCount = slicingMaelstromCount + 1
+	self:CDBar(args.spellId, 24, CL.count:format(args.spellName, slicingMaelstromCount))
```

Comment:
With the text added to the bar the old bar will no longer be overwritten if this ability happens early. Have you verified that Slicing Maelstrom can't happen before 23s on pull and every 24s after that?

If it's possible then a :StopBar can be used the text from the previous bar, or the timer could be updated to the minimum time.